dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/impl/ContextualProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.concurrent.internal.compat.impl;
+
+import java.util.ArrayList;
+import java.util.concurrent.Flow.Processor;
+import java.util.concurrent.Flow.Subscriber;
+import java.util.concurrent.Flow.Subscription;
+
+import com.ibm.ws.concurrent.ContextualAction;
+import com.ibm.wsspi.threadcontext.ThreadContext;
+import com.ibm.wsspi.threadcontext.ThreadContextDescriptor;
+
+/**
+ *
+ */
+public class ContextualProcessor<T, R> implements Processor<T, R>, ContextualAction<Processor<T, R>> {
+    private final Processor<T, R> action;
+    private final ThreadContextDescriptor threadContextDescriptor;
+
+    public ContextualProcessor(ThreadContextDescriptor threadContextDescriptor, Processor<T, R> action) {
+        this.action = action;
+        this.threadContextDescriptor = threadContextDescriptor;
+    }
+
+    @Override
+    public Processor<T, R> getAction() {
+        return action;
+    }
+
+    @Override
+    public ThreadContextDescriptor getContextDescriptor() {
+        return threadContextDescriptor;
+    }
```

Comment:
ContextualSubscriber includes the Trivial annotation to avoid tracing these. We should probably either add the same here, or remove it from ContextualSubscriber to make them consistent.

```suggestion
    @Override
    @Trivial
    public Processor<T, R> getAction() {
        return action;
    }

    @Override
    @Trivial
    public ThreadContextDescriptor getContextDescriptor() {
        return threadContextDescriptor;
    }
```

Reply:
Good catch, `@Trivial` is a good idea for these